src/librbd/Utils.cc

```diff
@@ -31,5 +39,38 @@ librados::AioCompletion *create_rados_ack_callback(Context *on_finish) {
   return create_rados_ack_callback<Context, &Context::complete>(on_finish);
 }
 
+int parse_rbd_default_features(CephContext* cct, const std::string& str_value, uint64_t* value) 
+{
+  int ret = 0;
+  *value = 0;
+  try {
+    *value = std::stoi(str_value, NULL, 10);
+  } catch (...) {
+    map<std::string, int> conf_vals = {{RBD_FEATURE_NAME_LAYERING, RBD_FEATURE_LAYERING}, 
+                                       {RBD_FEATURE_NAME_STRIPINGV2, RBD_FEATURE_STRIPINGV2},
+                                       {RBD_FEATURE_NAME_EXCLUSIVE_LOCK, RBD_FEATURE_EXCLUSIVE_LOCK},
+                                       {RBD_FEATURE_NAME_OBJECT_MAP, RBD_FEATURE_OBJECT_MAP},
+                                       {RBD_FEATURE_NAME_FAST_DIFF, RBD_FEATURE_FAST_DIFF},
+                                       {RBD_FEATURE_NAME_DEEP_FLATTEN, RBD_FEATURE_DEEP_FLATTEN},
+                                       {RBD_FEATURE_NAME_JOURNALING, RBD_FEATURE_JOURNALING}};
+    std::vector<std::string> strs;
+    boost::split(strs, str_value, boost::is_any_of(","));
+    for (std::vector<std::string>::const_iterator i = strs.begin(); i != strs.end(); ++i) {
+    	std::string feature = *i;
+    	boost::trim(feature);
+      if (conf_vals.find(feature) != conf_vals.end())
+        *value += conf_vals[feature];
```

Comment:
According to Ceph Coding style we try to follow Google's coding guide: https://google.github.io/styleguide/cppguide.html

And it states: "if one part of an if-else statement uses curly braces, the other part must too"